arelle/plugin/validate/NL/rules/fr_kvk.py

```diff
@@ -80,3 +82,104 @@ def rule_fr_kvk_2_01(
                     acceptedLangs=", ".join(ACCEPTED_LANGUAGES),
                     lang=lang,
                 )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+    ],
+)
+def rule_fr_kvk_2_03(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-KVK-2.03: The attribute 'href' of the 'link:schemaRef' element MUST refer to the
+    full web location of the published entrypoint from the annual reporting taxonomy
+    """
+    modelXbrl = val.modelXbrl
+    for doc in modelXbrl.urlDocs.values():
+        if doc.type == ModelDocument.Type.INSTANCE:
+            for refDoc, docRef in doc.referencesDocument.items():
+                if "href" not in docRef.referenceTypes:
+                    continue
+                if docRef.referringModelObject.localName != "schemaRef":
+                    continue
+                href = refDoc.uri
+                ext = None
+                if href:
+                    path = urlparse(href.lower()).path
+                    ext = os.path.splitext(path)[1]
+                if not UrlUtil.isAbsolute(href) or ext != '.xsd':
```

Comment:
It seems like the disclosure system should have the set of supported set of XSDs for that version rather than just heuristically checking the shape of the URI.  e.g., EFM checks for specific URLs to ensure you're not referencing something too old.

Reply:
I've just pushed a fairly simple solution, but I did notice that EFM uses a more complicated [XML format](https://github.com/Arelle/Arelle/blob/72aa068e8b7e53d59f1a3f3d371f7fec25a994f2/arelle/plugin/validate/EFM/resources/edgartaxonomies/edgartaxonomies-23-3.xml) to [define attributes](https://github.com/Arelle/Arelle/blob/72aa068e8b7e53d59f1a3f3d371f7fec25a994f2/arelle/DisclosureSystem.py#L272-L326) associated with entry points. It seems fairly specific to EFM, I didn't see any other built-in validation plugins making use of this design, and I don't see any immediate value-add of using that design over this simple solution.

Reply:
The advantage is `<schemaRef href="http://mycompany.com/myextensiontaxonomy.xsd"/>` passes your filter, but the rule says it shouldn't.  The rule says you must use a pre-approved entry points, not an arbitrary one that happens to be an absolute URL that ends with `.xsd`.

Reply:
I pushed a [simple solution](https://github.com/Arelle/Arelle/pull/875/commits/b459e9816fe1bbb516593e886de18e0fb9c8c1aa) (but not as simple as my initial solution) with my last comment that does check against pre-approved entry points. That comment was more about whether or not we should make use of the same XML-based approach that EFM uses, or if this simple solution checking against a predefined list is sufficient for this validation.

Reply:
Oh, sorry for the misunderstand.  I have no idea, sorry :-).